cmd/prometheus/main.go

```diff
@@ -252,6 +258,11 @@ func main() {
 		newFlagRetentionDuration model.Duration
 	)
 
+	// Unregister the default GoCollector, and reregister with our defaults.
+	if prometheus.Unregister(collectors.NewGoCollector()) {
```

Comment:
Kinda cruddy that we have to do this, but it seems forced from [here](https://github.com/prometheus/client_golang/blob/18b51345cdbaea834b49ea5303e67e15e54d1431/prometheus/registry.go#L62) and Prometheus' use of the defaultRegistry.

Reply:
Yup, we could plumb in a non-default registry in `main`. But that would be a bit more work. This works fine for now, we could open an issue for improvement.